internal/state/graph/graph.go

```diff
@@ -9,12 +9,12 @@ import (
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state/validation"
 )
 
-// ClusterStore includes cluster resources necessary to build the Graph.
-type ClusterStore struct {
-	GatewayClass *v1beta1.GatewayClass
-	Gateways     map[types.NamespacedName]*v1beta1.Gateway
-	HTTPRoutes   map[types.NamespacedName]*v1beta1.HTTPRoute
-	Services     map[types.NamespacedName]*v1.Service
+// ClusterState includes cluster resources necessary to build the Graph.
+type ClusterState struct {
+	GatewayClasses map[types.NamespacedName]*v1beta1.GatewayClass
```

Comment:
This simplifies the implementation of `ChangeProcessor`.

Note that `changeTrackingUpdater` works with `multiObjectStore` which works with `objectStore`s which work with maps.  

Using GatewayClass field instead of `GatewayClasses map[types.NamespacedName]*v1beta1.GatewayClass` in ClusterState means there will be a need for an additional logic in `ChangeProcessor` to convert the map (used by objectStore) to a single resource. It is easier if we have the map in the first place, so we don't need to do any convertion.